### PR TITLE
Upstream merge 05 18 22

### PR DESCRIPTION
**Pull request by bartgol**
There were a four conflicts. The xml files conflicts were relatively easy. In all the XML files, I picked upstream for the non-scream compsets, and HEAD for the scream compsets.

The toughest conflict was in `components/eam/src/physics/cam/check_energy.F90`. @oksanaguba and @AaronDonahue might want to take a look at that. Oksana just merged a PR on energy in the upstream, that reorganizes/changes the implementation of that module. In the version in the scream repo, I saw comments by Aaron regarding P3 vs MG. I ended up picking the upstream version completely, but I don't know if this messed up things for scream v0?

@ndkeen I added you too so you can make sure the important PM mods made it in.

### CODE REVIEW

**Review by ndkeen — DISMISSED** [review]
Just quickly, all the perlmutter-related changes look fine.

**Comment by bartgol** [comment]
@AaronDonahue I assume the change in `check_energy.F90` caused some issues in v0. Would you mind taking a look at the file, and figure out how to merge  the upstream version with the scream one?

File: components/eam/src/physics/cam/check_energy.F90 [hunk 1]
```diff
@@ -1308,6 +1111,125 @@ subroutine check_prect(state, tend, name, nstep, ztodt, prect)
...
+    se = se + phis*ps/gravit
+
+    do k = 1, pver
+       wl = wl + q(k,icldliq)*pdel(k)/gravit
+       wi = wi + q(k,icldice)*pdel(k)/gravit
+    end do
+
+    do k = 1, pver
+       wr = wr + q(k,irain)*pdel(k)/gravit
+       ws = ws + q(k,isnow)*pdel(k)/gravit
```
Comment (AaronDonahue) [inline components/eam/src/physics/cam/check_energy.F90:1223]: We probably need to add a conditional here... I'll add the change.

Comment (bartgol) [reply components/eam/src/physics/cam/check_energy.F90:1223]: Thanks!

**Comment by AaronDonahue** [comment]
@bartgol , I'll let the AT verify this fix works.  I just pushed a change to check_energy that will hopefully cause SCREAMv0 to pass now.

**Comment by bartgol** [comment]
We got still fails in v0. The test `SMS_R_Ln10.ne4_ne4.FDPSCREAMA97` failed with error

```
 ERROR: SCAM_SETOPTS: SCMLAT must be between -90. and 90. degrees.
```

**Comment by PeterCaldwell** [comment]
That's an error for @bogensch

**Comment by bogensch** [comment]
DP-SCREAM tests should pass now (at least they do for me on LC now).

**Comment by bartgol** [comment]
@AaronDonahue it looks like there is still some issue with the check_energy module:

```
Error termination. Backtrace:
At line 1167 of file /home/e3sm-jenkins/jenkins-ws/workspace/SCREAM_PullRequest_Autotester_Mappy/2574/scream/components/eam/src/physics/cam/check_energy.F90
Fortran runtime error: Index '9' of dimension 1 of array 'ws' above upper bound of 8
```
The error appears to impact only the DEBUG test cases.

**Comment by AaronDonahue** [comment]
> @AaronDonahue it looks like there is still some issue with the check_energy module:
> 
> ```
> Error termination. Backtrace:
> At line 1167 of file /home/e3sm-jenkins/jenkins-ws/workspace/SCREAM_PullRequest_Autotester_Mappy/2574/scream/components/eam/src/physics/cam/check_energy.F90
> Fortran runtime error: Index '9' of dimension 1 of array 'ws' above upper bound of 8
> ```
> 
> The error appears to impact only the DEBUG test cases.

@bartgol , okay this latest commit should fix this.  I accidentally put `ws(i)` outside of the `for i` loop.

**Comment by bartgol** [comment]
For the sake of avoiding time outs (other folks are using mappy), and since the standalone tests are passing, I'm making the AT skip standalone tests nest round. Hopefully will be enough to get a full v0/v1 run.

**Comment by bartgol** [comment]
Ok, the fails are expected (we need to bless tonights nightlies).

**Review by AaronDonahue — APPROVED** [review]

**Comment by bartgol** [comment]
The standalone tests on mappy had some odd failures. Looking at the log, all REQUIRE statements passed, but mpirun returned nonzero exit codes. Might be some issue with cleanup, although the logs don't show any error whatsoever.

**Comment by AaronDonahue** [comment]
The most recent set of AT tests looks like the mappy standalone got figured out.  But unfortunately it's still failing in FDPSCREAM97.  @bogensch , do you have any thoughts on why this test may be failing now?

The other "fails" are NL diffs, which I think we are okay with right?

**Comment by bogensch** [comment]
These tests are all passing for me on LC and Cori after my fix I pushed.  Any chance I can get a more verbose error?

**Comment by bartgol** [comment]
The FAIL was due to bad compset name, it appers:
```
ERROR: Invalid compset name, FDPSCREAMA97, all stub components generated
```

**Comment by bogensch** [comment]
The test is now named "FDPSCREAM-ARM97".  I modified this name in cime_config/tests.py (https://github.com/E3SM-Project/scream/pull/1649/commits/996040743feb7e25180b68c41871837f08fb3935), so I'm confused why it's trying to run "FDPSCREAMA97".  does the autotester need to be updated to reflect the change or something?

**Comment by bartgol** [comment]
It probably got messed up in the merge. Other PRs also changed tests.py recently, so that's probably my fault in fixing the conflicts.

**Comment by bartgol** [comment]
Besides the weaver fail (irrelevant, since changes since last weaver pass only affected CIME cases), there are 4 NLFAIL and 1 DIFF in v0. The diff is due to missing baselines (the compset name changed). The NLFAILs are all the same

```
  BASE: se_ne_x = -1
  COMP: se_ne_x = 0 
  BASE: se_ne_y = -1
  COMP: se_ne_y = 0 
```
which are expected, I think. So as far as testing is concerned, this PR is ready.

**Comment by bartgol** [comment]
@AaronDonahue ready for your review, if you want.

**Review by AaronDonahue — APPROVED** [review]

**Comment by AaronDonahue** [comment]
I think this is ready to go.  @bartgol , can you confirm the only fails we are getting are from NLFAIL?  Which are expected right?  If so, go ahead and manually merge.

**Comment by bartgol** [comment]
Yes. NLFAIL and a diff due to a compset name change (hence, missing baseline)

**Pull request merged**